lib/reporters.go

```diff
@@ -109,3 +109,134 @@ func NewJSONReporter(m *Metrics) Reporter {
 		return json.NewEncoder(w).Encode(m)
 	}
 }
+
+var hdrHeader = []byte("Value    Percentile    TotalCount    1/(1-Percentile)\n\n")
+
+var logarithmic = []float64{
+	0.00,
+	0.100,
+	0.200,
+	0.300,
+	0.400,
+	0.500,
+	0.550,
+	0.600,
+	0.650,
+	0.700,
+	0.750,
+	0.775,
+	0.800,
+	0.825,
+	0.850,
+	0.875,
+	0.8875,
+	0.900,
+	0.9125,
+	0.925,
+	0.9375,
+	0.94375,
+	0.950,
+	0.95625,
+	0.9625,
+	0.96875,
+	0.971875,
+	0.975,
+	0.978125,
+	0.98125,
+	0.984375,
+	0.985938,
+	0.9875,
+	0.989062,
+	0.990625,
+	0.992188,
+	0.992969,
+	0.99375,
+	0.994531,
+	0.995313,
+	0.996094,
+	0.996484,
+	0.996875,
+	0.997266,
+	0.997656,
+	0.998047,
+	0.998242,
+	0.998437,
+	0.998633,
+	0.998828,
+	0.999023,
+	0.999121,
+	0.999219,
+	0.999316,
+	0.999414,
+	0.999512,
+	0.999561,
+	0.999609,
+	0.999658,
+	0.999707,
+	0.999756,
+	0.99978,
+	0.999805,
+	0.999829,
+	0.999854,
+	0.999878,
+	0.99989,
+	0.999902,
+	0.999915,
+	0.999927,
+	0.999939,
+	0.999945,
+	0.999951,
+	0.999957,
+	0.999963,
+	0.999969,
+	0.999973,
+	0.999976,
+	0.999979,
+	0.999982,
+	0.999985,
+	0.999986,
+	0.999988,
+	0.999989,
+	0.999991,
+	0.999992,
+	0.999993,
+	0.999994,
+	0.999995,
+	0.999996,
+	0.999997,
+	0.999998,
+	0.999999,
+	1.0,
+}
+
+// NewHDRHistogramPlotReporter returns a Reporter that writes out latency metrics
+// in a format plottable by http://hdrhistogram.github.io/HdrHistogram/plotFiles.html.
+func NewHDRHistogramPlotReporter(m *Metrics) Reporter {
+	return func(w io.Writer) error {
+		if _, err := w.Write(hdrHeader); err != nil {
+			return err
+		}
+
+		const fmtstr = "%f    %f        %d            %f\n"
+		total := m.Requests
+
+		for _, q := range logarithmic {
+			value := m.Latencies.Quantile(q).Seconds() * 1000 // milliseconds
```

Comment:
This goes through an extra unnecessary float operation (ns -> s -> ms) (potentially introducing errors. might not happen for our number ranges, not sure), and also doesn't use the useful `time` constants. Why not:

```
 int64(m.Latencies.Quantile(q) / time.Millisecond)
```

Reply:
We don't want a integer rounding so I used the float conversion method the stdlib uses, but for milliseconds.